solr/core/src/java/org/apache/solr/cli/PostTool.java

```diff
@@ -315,8 +315,7 @@ public void runImpl(CommandLine cli) throws Exception {
         throw new IllegalArgumentException(
             "Must specify -c / --name parameter with --solr-url to post documents.");
       }
-      String url =
-          SolrCLI.normalizeSolrUrl(cli) + "/solr/" + cli.getOptionValue("name") + "/update";
+      String url = SolrCLI.normalizeSolrUrl(cli) + "/" + cli.getOptionValue("name") + "/update";
```

Comment:
Cannot always blindly add `/solr`. The tests use randomized `contextPath` which may be `/xy/zx/` for a running Jetty, and then the test will pass in a `--solr-url` such as `http://localhost:9999/xy/zx/`, and the normalization will then return the same path which is correct. Question is whether the `normalizeSolrUrl` command should append `/solr` if no context path is given at all, the same way that `SolrCLI.getSolrClient(cli)` does.

Reply:
I think, and I'm tagging @gerlowskija as well, that the overall idea is that we want bare Solr urls, and then add the appropriate `/solr` or `/api` as needed.

Reply:
That's how it is on main. But the "`/` as baseUrl" code is not (fully) backPorted to 9x, so there are differences. In 99% of cases our context path is `/solr` but some tests spin up Jetty's with other paths and they will break if we start creeping in too many hardcoded `/solr`. Should be safe to do in a test here and there though.

Reply:
Oh right..  It was 10x only change in SOLR-16911 that did this.